test/test_verifylib.py

```diff
@@ -156,121 +159,275 @@ def test_commands_do_not_align_at_all_log_warning(self):
+      verify_create_rule(rule, materials_queue, products_queue)
 
-  def test_pass_ignore_case_keyword(self):
-    """["delete", "bar"], ["DELETE", "bar"], ignores keyword case, passes. """
+    # Not all (*) products newly created
+    materials_queue = ["foo"]
+    products_queue = ["foo", "bar"]
+    rule = ["CREATE", "*"]
+    with self.assertRaises(RuleVerficationError):
+      verify_create_rule(rule, materials_queue, products_queue)
 
-    rule1 = ["delete", "bar"]
-    rule2 = ["DELETE", "bar"]
-    self.assertIsNone(
-        verify_delete_rule(rule1, self.artifact_queue))
-    self.assertIsNone(
-        verify_delete_rule(rule2, self.artifact_queue))
+  def test_pass(self):
+    """"Different scenarios for passing create rule verification. """
+    # Foo created
+    materials_queue = ["bar"]
+    products_queue = ["foo", "bar"]
+    rule = ["CREATE", "foo"]
+    queue = verify_create_rule(rule, materials_queue, products_queue)
+    self.assertListEqual(queue, ["bar"])
 
+    # * created
+    materials_queue = []
+    products_queue = ["foo", "bar", "baz"]
+    rule = ["CREATE", "*"]
+    queue = verify_create_rule(rule, materials_queue, products_queue)
+    self.assertListEqual(queue, [])
 
+    # No products filtered by pattern
+    materials_queue = ["foo", "bar"]
+    products_queue = []
+    rule = ["CREATE", "*"]
+    queue = verify_create_rule(rule, materials_queue, products_queue)
+    self.assertListEqual(queue, [])
 
+    # No products filtered by pattern (pass seems strange)
+    materials_queue = ["foo", "bar"]
+    products_queue = []
+    rule = ["CREATE", "baz"]
+    queue = verify_create_rule(rule, materials_queue, products_queue)
+    self.assertListEqual(queue, [])
 
 
-class TestVerifyCreateRule(unittest.TestCase):
-  """Test verifylib.verify_create_rule(rule, artifact_queue) """
+class TestVerifyModifyRule(unittest.TestCase):
+  """Test verifylib.verify_modify_rule
+  takes a rule ["MODIFY", "<path pattern>"], a product queue, a
+  material queue, a material dict and a product dict.
 
-  def setUp(self):
-    """ Setup artifact queues. """
-    self.artifact_queue = ["foo"]
-    self.artifact_queue_foostar = ["foo", "bar", "foobar"]
-    self.artifact_queue_empty = []
+  The sets of materials and products from the queue filtered by the path pattern
+  must be equal in terms of paths.
+  And each material-product must have different hashes.
 
-  def test_fail_create_file(self):
-    """["CREATE", "bar"], does not mach bar (not created), fails. """
+  Returns updated materials and products queues minus the respective filtered
+  artifacts.
+  """
 
-    rule = ["CREATE", "bar"]
-    with self.assertRaises(RuleVerficationError):
-      verify_create_rule(rule, self.artifact_queue)
+  @classmethod
+  def setUpClass(self):
+    sha_1 = ("d65165279105ca6773180500688df4bd"
+                  "c69a2c7b771752f0a46ef120b7fd8ec3")
 
+    sha_2 = ("155c693a6b7481f48626ebfc545f0523"
```

Comment:
do we want to change these names to something that doesn't hint of either algorithm?